scripts/flash

```diff
@@ -1,10 +1,49 @@
 #!/bin/bash
 
-FIRMWARE_NAME=konnected-firmware-2-3-0.bin
-FILESYSTEM_NAME=konnected-filesystem-0x100000-2-3-0.img
-PORT=/dev/cu.wchusbserial1410
+function usage() {
+    echo
+    echo "usage: $0 <version> <port>"
+    echo
+    echo '    version should be formatted <major>-<minor>-<patch>'
+    echo
+    echo 'Execute from the root dir of the repo.'
+    echo "    ex: '$0 /dev/ttyS5 2-3-0'"
+    exit 1
+}
+if [ "$#" == 0 ]; then
+    usage
+fi
 
-esptool.py --port=${PORT} write_flash --flash_mode dio 0x00000 firmware/${FIRMWARE_NAME}
-esptool.py --port=${PORT} write_flash --flash_mode dio 0x100000 firmware/${FILESYSTEM_NAME}
+VERSION="$1"
+if ! [[ "${VERSION}" =~ ^[0-9]+-[0-9]+-[0-9]+ ]] ; then
+    echo "Version not formatted correctly..."
+    usage
+fi
 
-tput bel
\ No newline at end of file
+PORT=${2:-/dev/cu.wchusbserial1410}
+BUILD_PATH="firmware/builds/${VERSION}"
+
+if [ ! -d "${BUILD_PATH}" ] ; then
+    echo "No matching build version - trying releases"
+    BUILD_PATH="firmware/releases/${VERSION}"
+fi
+
+FIRMWARE_NAME="${BUILD_PATH}/konnected-firmware-${VERSION}.bin"
+FILESYSTEM_NAME="${BUILD_PATH}/konnected-filesystem-0x100000-${VERSION}.img"
+
+if ! [ -f ${FIRMWARE_NAME} ] ; then
+    echo "Firmware build not found...expected at '${FIRMWARE_NAME}'"
+    usage
+fi
+
+if ! [ -f ${FILESYSTEM_NAME} ] ; then
+    echo "Filesystem build not found...expected at '${FILESYSTEM_NAME}'"
+    usage
+fi
+
+esptool.py "--port=${PORT}" write_flash --flash_mode dio 0x00000 "${FIRMWARE_NAME}"
+esptool.py "--port=${PORT}" write_flash --flash_mode dio 0x100000 "${FILESYSTEM_NAME}"
```

Comment:
To make this even better and more generic we could parse the flash address from the filename. Not really necessary but just an idea.

Reply:
Good idea and easy add - will update